numba/core/overload_glue.py

```diff
@@ -0,0 +1,184 @@
+"""
+Provides wrapper functions for "glueing" together Numba implementations that are
+written in the "old" style of a separate typing and lowering implementation.
+"""
+import types as pytypes
+import textwrap
+
+
+class _OverloadWrapper(object):
+    """This class does all the work of assembling and registering wrapped split
+    implementations.
+    """
+
+    def __init__(self, function=None):
```

Comment:
why default to None when the next line is forbidding it?

Reply:
I don't think it needs to be a kwarg any more.